dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/datavalueset/DefaultDataValueSetService.java

```diff
@@ -629,14 +630,33 @@ public ImportSummary importDataValueSetPdf( InputStream in, ImportOptions option
     private ImportSummary importDataValueSet( ImportOptions options, JobConfiguration id,
         Callable<DataValueSetReader> createReader )
     {
+        options = ObjectUtils.firstNonNull( options, ImportOptions.getDefaultImportOptions() );
+
+        BatchHandler<DataValue> dvBatch = batchHandlerFactory.createBatchHandler( DataValueBatchHandler.class );
+        BatchHandler<DataValueAudit> dvaBatch = batchHandlerFactory.createBatchHandler(
+            DataValueAuditBatchHandler.class );
+
+        notifier.clear( id );
+
         try ( DataValueSetReader reader = createReader.call() )
         {
-            return importDataValueSet( options, id, reader );
+            ImportSummary summary = importDataValueSet( options, id, reader, dvBatch, dvaBatch );
+
+            dvBatch.flush();
+            dvaBatch.flush();
+
+            NotificationLevel notificationLevel = options.getNotificationLevel( INFO );
+            notifier.notify( id, notificationLevel, "Import done", true )
+                .addJobSummary( id, notificationLevel, summary, ImportSummary.class );
+
+            return summary;
         }
         catch ( Exception ex )
         {
+            dvBatch.flush();
+            dvaBatch.flush();
```

Comment:
Would it be worth logging that the batch handlers are being flushed? (in the try and/or catch)
This would help when reading logs and not having to inspect the code.
Exposing app behaviour in the logs usually helps understand what the app is doing.

Reply:
Yes, in hindsight I learned that some more logging would be good here. As I said in the meeting the logging situation of the import needs general improvement. I make a new ticket for specifically working on the logging.

Reply:
Created https://dhis2.atlassian.net/browse/DHIS2-15396